sdk/core/src/context.rs

```diff
@@ -6,15 +10,84 @@
 /// pipeline execution history between policies.
 /// For example, it could be used to signal that an execution failed because a CosmosDB endpoint is
 /// down and the appropriate policy should try the next one).
-#[derive(Clone)]
+#[derive(Clone, Debug, Default)]
 pub struct Context {
-    // Temporary hack to make sure that Context is not initializeable
-    // Soon Context will have proper data fields
-    _priv: (),
+    property_bag: HashMap<TypeId, Arc<dyn Any + Send + Sync>>,
 }
 
 impl Context {
     pub fn new() -> Self {
-        Self { _priv: () }
+        Self::default()
+    }
+
+    pub fn property_bag(&self) -> &HashMap<TypeId, Arc<dyn Any + Send + Sync>> {
+        &self.property_bag
+    }
+
+    // **IMPORTANT**: Care must be taken to avoid storing a mismatched TypeId because the other,
+    // public functions, assume that the same TypeId used as key will be the one extracted from the
+    // hashmap. This is guaranteed by the public `set_property` function.
+    fn property_bag_mut(&mut self) -> &mut HashMap<TypeId, Arc<dyn Any + Send + Sync>> {
```

Comment:
Why does this function exist? If it's hard to use correctly and it's only being used one place, can we just inline it?

Reply:
Good idea. I wrote is as function before making it private and left it there as documentation. I suppose we should move the docs to the struct level.